services/wallet/controllers_v3.go

```diff
@@ -817,6 +818,20 @@ func DisconnectCustodianLinkV3(s *Service) func(w http.ResponseWriter, r *http.R
 	}
 }
 
+func isAllowedOrigin(origin string, allowedOrigins []string) bool {
```

Comment:
this looks good to me. Something to consider during deployment is to order this array so the first index is `rewards.brave.com` so that we don't end up unnecessarily looping.